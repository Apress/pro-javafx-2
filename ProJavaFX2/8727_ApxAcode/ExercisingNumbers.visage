/*
 * @test
 * @run
 */

var x = 3.14159;
var y = 2.71828;

println("x = {x}");
println("y = {y}");

println("x + y = {x + y}");
println("x - y = {x - y}");
println("x * y = {x * y}");
println("x / y = {x / y}");
println("x mod y = {x mod y}");

var max = Float.MAX_VALUE;
var min = Float.MIN_VALUE;

println("max = {max}");
println("min = {min}");

println("max * 2 will overflow to {max * 2}");
println("min / 2 will underflow to {min / 2}");

var inf = max * max;

println("inf = {inf}");
println("inf.isInfinite() = {inf.isInfinite()}");
println("inf / inf = {inf / inf}");

var nan = inf / inf;

println("nan = {nan}");
println("nan.isNaN() = {nan.isNaN()}");
