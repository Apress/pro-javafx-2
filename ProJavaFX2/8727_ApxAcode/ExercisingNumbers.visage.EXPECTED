x = 3.14159
y = 2.71828
x + y = 5.85987
x - y = 0.42331004
x * y = 8.5397215
x / y = 1.1557271
x mod y = 0.42331004
max = 3.4028235E38
min = 1.4E-45
max * 2 will overflow to Infinity
min / 2 will underflow to 0.0
inf = Infinity
inf.isInfinite() = true
inf / inf = NaN
nan = NaN
nan.isNaN() = true
